dev/tools/controllerbuilder/pkg/llm/gemini.go

```diff
@@ -150,14 +152,38 @@ func toGenaiSchema(schema *Schema) (*genai.Schema, error) {
 // 	})
 // }
 
+func (c *GeminiChat) retryWithJitter(ctx context.Context, geminiParts ...genai.Part) (*genai.GenerateContentResponse, error) {
+	for {
+		geminiResponse, err := c.chat.SendMessage(ctx, geminiParts...)
+
+		if err == nil {
+			return geminiResponse, nil
+		}
+		if !IsResourceExhausted(err) {
+			return nil, err
+		}
+
+		backoff := gax.Backoff{
```

Comment:
If you initialize backoff inside the loop, do we actually backoff?

Reply:
😮‍💨  my brain was dumb. Also I don't like this for loop. I rewrote completely.  Let me try it out before merging the code.